src/server/game/Entities/Unit/Unit.h

```diff
@@ -1686,6 +1686,11 @@ class TC_GAME_API Unit : public WorldObject
 
         float GetCollisionHeight() const override;
 
+        // returns if the unit is ignoring any combat interaction
+        bool IsIgnoringCombat() const { return _isIgnoringCombat; }
```

Comment:
jfyi: could it better to do smth like `return _isIgnoringCombat || flags_extra & CREATURE_FLAG_EXTRA_NO_COMBAT` instead of a getter function?

Reply:
No, because this state is suposed to be modifyable during runtime. A flag check would defy that purpose because extra flags are static and cannot be changed.